qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -187,39 +195,195 @@ def run(self,
                 DeprecationWarning, stacklevel=2)
             if parameter_binds:
                 raise AerError("Parameter binds can't be used with an input qobj")
-            # A work around to support both qobj options and run options until
-            # qobj is deprecated is to copy all the set qobj.config fields into
-            # run_options that don't override existing fields. This means set
-            # run_options fields will take precidence over the value for those
-            # fields that are set via assemble.
-            if not run_options:
-                run_options = circuits.config.__dict__
+            qobj = circuits
+        else:
+            if not isinstance(circuits, list):
+                circuits = [circuits]
+
+            if ((hasattr(self._options, 'executor') and self._options.executor) or
+                    'executor' in run_options and run_options['executor']):
+                for circuit in circuits:
+                    if not isinstance(circuit, (QuantumCircuit, Schedule)):
+                        raise AerError(f"Unsupported circuit for simulation: {circuit.__class__}")
+                qobj = self._assemble(circuits, parameter_binds=parameter_binds, **run_options)
             else:
-                run_options = copy.copy(run_options)
-                for key, value in circuits.config.__dict__.items():
-                    if key not in run_options and value is not None:
-                        run_options[key] = value
-            qobj = self._assemble(circuits, **run_options)
+                def is_conditional_circuit(circuit):
+                    if isinstance(circuit, AerCircuit):
+                        return False
+                    for inst, _, _ in circuit.data:
+                        if inst.condition:
+                            return True
+                    return False
+                if (not all(isinstance(c, (QuantumCircuit, AerCircuit)) for c in circuits) or
+                        any(is_conditional_circuit(c) for c in circuits) or parameter_binds):
```

Comment:
circuits with conditional instructions are routed to the path of `_submit_qobj` currently. After supporting `bFunc`, all the circuits will be routed to the path of `_submit_circuits`.